aws-transfer-user/src/test/java/software/amazon/transfer/user/UpdateHandlerTest.java

```diff
@@ -0,0 +1,236 @@
+package software.amazon.transfer.user;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static software.amazon.transfer.user.BaseHandlerStd.THROTTLE_CALLBACK_DELAY_SECONDS;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.services.transfer.model.DescribeUserRequest;
+import software.amazon.awssdk.services.transfer.model.DescribeUserResponse;
+import software.amazon.awssdk.services.transfer.model.TagResourceRequest;
+import software.amazon.awssdk.services.transfer.model.TagResourceResponse;
+import software.amazon.awssdk.services.transfer.model.ThrottlingException;
+import software.amazon.awssdk.services.transfer.model.UntagResourceRequest;
+import software.amazon.awssdk.services.transfer.model.UntagResourceResponse;
+import software.amazon.awssdk.services.transfer.model.UpdateUserRequest;
+import software.amazon.awssdk.services.transfer.model.UpdateUserResponse;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+@ExtendWith(MockitoExtension.class)
+public class UpdateHandlerTest extends AbstractTestBase {
+
+    final UpdateHandler handler = new UpdateHandler();
+
+    @Test
+    public void handleRequest_SimpleSuccess() {
+        ResourceModel model = simpleUserModel();
+        updateAndAssertSuccess(model, model);
+    }
+
+    @Test
+    public void handlerRequest_VerifyAddingAndRemovingKeys() {
+        String key1 = "ssh-rsa abcd";
+        String key2 = "ssh-rsa efgh";
+        String key3 = "ssh-rsa ijklm";
+
+        ResourceModel current = simpleUserModel();
+        current.setSshPublicKeys(Collections.singletonList(key1));
+
+        ResourceModel desired = simpleUserModel();
+        desired.setSshPublicKeys(Arrays.asList(key1, key2, key3));
+
+        updateAndAssertSuccess(current, desired);
+
+        current.setSshPublicKeys(Arrays.asList(key1, key2, key3));
+        desired.setSshPublicKeys(Collections.singletonList(key1));
```

Comment:
Nit: can we use `List.of` or is this targetting Java 8?

Reply:
We are targeting Java 11 from what I recall. But I need to check if the target bytecode settings are reflecting this.